splice_si_api/src/main/java/com/splicemachine/si/impl/store/IgnoreTxnSupplierImpl.java

```diff
@@ -87,7 +95,35 @@ private void populateIgnoreTxnCache() throws IOException {
                     }
                 }
             }
+            cache = combineOverlappingRanges(cache);
+        }
+    }
+
+    static Set<Pair<Long, Long>> combineOverlappingRanges(Set<Pair<Long, Long>> cache) {
```

Comment:
I don't expect this to be a problem initially, but the algorithm could blow up since it has quadratic complexity. Add some logging information when populating the cache so that if we run into this problem at least it's clear where we are spending too much time.

Reply:
Added logs in commit https://github.com/splicemachine/spliceengine/pull/5103/commits/19138b4bb9ba883e90b87b67324df4a83a7fd959

Reply:
I think we can reduce the complexity of the overlapping algorithm by maintaining the cache online, i.e. whenever we add a new element, we change the cache immediately, provided that we have a suitable equality / hash function and we use a TreeMap, we can probably reduce the complexity to from quadratic to logarithmic. If there is a JIRA open for this, please refer me to it so we can follow up there.